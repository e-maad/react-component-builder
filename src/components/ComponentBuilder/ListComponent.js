import React from 'react';

const ListComponent = ({ li, id }) => {
    return <ul id={id} className='list-component'>
        {li ? li.map(item => <li key={item}>{item}</li>) : ''}
    </ul>
}

export default ListComponent;