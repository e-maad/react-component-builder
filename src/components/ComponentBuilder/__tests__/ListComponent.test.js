import React from 'react';
import ListComponent from '../ListComponent';


const wrapper = shallow(<ListComponent li={[
    "free to use",
    "superfast",
    "and pretty, too!"
]} />);

describe('List Component', () => {

    it('match the snapshot', () => {
        const wrapper = renderer.create(<ListComponent />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('base element should be present', () => {
        expect(wrapper.find('.list-component').length).toBe(1);
    });
});