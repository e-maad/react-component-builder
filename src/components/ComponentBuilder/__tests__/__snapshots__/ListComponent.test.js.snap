// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component match the snapshot 1`] = `
<ul
  className="list-component"
>
  
</ul>
`;
